site/src/pages/DeploySettingsPage/NotificationsPage/NotificationEvents.tsx

```diff
@@ -0,0 +1,224 @@
+import type { Interpolation, Theme } from "@emotion/react";
+import Button from "@mui/material/Button";
+import Card from "@mui/material/Card";
+import Divider from "@mui/material/Divider";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemText, { listItemTextClasses } from "@mui/material/ListItemText";
+import ToggleButton from "@mui/material/ToggleButton";
+import ToggleButtonGroup from "@mui/material/ToggleButtonGroup";
+import Tooltip from "@mui/material/Tooltip";
+import {
+	type selectTemplatesByGroup,
+	updateNotificationTemplateMethod,
+} from "api/queries/notifications";
+import type { DeploymentValues } from "api/typesGenerated";
+import { Alert } from "components/Alert/Alert";
+import { displaySuccess } from "components/GlobalSnackbar/utils";
+import { Stack } from "components/Stack/Stack";
+import {
+	type NotificationMethod,
+	castNotificationMethod,
+	methodIcons,
+	methodLabels,
+} from "modules/notifications/utils";
+import { type FC, Fragment } from "react";
+import { useMutation, useQueryClient } from "react-query";
+import { docs } from "utils/docs";
+
+type NotificationEventsProps = {
+	defaultMethod: NotificationMethod;
+	availableMethods: NotificationMethod[];
+	templatesByGroup: ReturnType<typeof selectTemplatesByGroup>;
+	deploymentValues: DeploymentValues;
+};
+
+export const NotificationEvents: FC<NotificationEventsProps> = ({
+	defaultMethod,
+	availableMethods,
+	templatesByGroup,
+	deploymentValues,
+}) => {
+	const hasWebhookNotifications = Object.values(templatesByGroup)
+		.flat()
+		.some((t) => t.method === "webhook");
+	const hasEmailNotifications = Object.values(templatesByGroup)
+		.flat()
+		.some((t) => t.method === "smtp");
+
+	return (
+		<Stack spacing={4}>
+			{hasWebhookNotifications &&
+				deploymentValues.notifications?.webhook.endpoint === "" && (
+					<Alert
+						severity="warning"
+						actions={
+							<Button
+								variant="text"
+								size="small"
+								component="a"
+								target="_blank"
+								rel="noreferrer"
+								href={docs("/cli/server#--notifications-webhook-endpoint")}
+							>
+								Read the docs
+							</Button>
+						}
+					>
+						Webhook notifications are enabled, but no endpoint has been
+						configured.
+					</Alert>
+				)}
+
+			{hasEmailNotifications &&
+				deploymentValues.notifications?.email.smarthost === "" && (
+					<Alert
+						severity="warning"
+						actions={
+							<Button
+								variant="text"
+								size="small"
+								component="a"
+								target="_blank"
+								rel="noreferrer"
+								href={docs("/cli/server#--notifications-email-smarthost")}
+							>
+								Read the docs
+							</Button>
+						}
+					>
+						SMTP notifications are enabled, but no smarthost has been
```

Comment:
Ditto, and please include the other 2 required fields:
https://coder.com/docs/admin/notifications#smtp-email

Reply:
Why not omit the SMTP option from the available methods endpoint when the required fields are not set? As a user, it's confusing to see SMTP listed as an available method when it's not truly available to be used. What do you think?

Reply:
Won't that require a lot of refactoring of what you have here? Plus what we do for SMTP we must also do for webhooks.

Plus available != configured. SMTP is always available, but unless it's configured correctly it's not usable.

Reply:
> Plus available != configured. SMTP is always available, but unless it's configured correctly it's not usable.

Hm... I see. Thanks for explaining it.

Reply:
I'm going to update the copy to "SMTP notifications are enabled but not properly configured." Wdyt? It is more general and we don't have to point property by property in the message since we already link to the docs.